### PR TITLE
resurface lost ddp info message

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

The following message disappeared from Lightning:

```
----------------------------------------------------------------------------------------------------
distributed_backend=ddp
All DDP processes registered. Starting ddp with 2 processes
----------------------------------------------------------------------------------------------------
```

The condition for printing it was mistakenly `if not torch.distributed.is_initialized`, which would never be true since right before that we initialized the backend. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]
thanks for the quick fix!

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 1]
```diff
@@ -308,6 +301,12 @@ def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Opt
             log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
+            # on rank=0 let everyone know training is starting
+            rank_zero_info("-" * 100)
+            rank_zero_info(f"distributed_backend={self.distributed_backend}")
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/ddp.py:306]: Should this say accelerator since it's the argument the users set?

```suggestion
            rank_zero_info(f"accelerator={self.distributed_backend}")
```

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp.py:306]: good point, or is it maybe supposed to say

`self.torch_distributed_backend`, e.g., either gloo or nccl?

now actually very unsure what is meant here ... hmm.

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 2]
```diff
@@ -308,6 +301,12 @@ def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Opt
...
+            rank_zero_info(f"All DDP processes registered. Starting ddp with {self.world_size} processes")
+            rank_zero_info("-" * 100)
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/ddp.py:308]: can we rather form it as one long message instead of 4x calls?

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp.py:308]: can you show how?
the string concatenation with line breaks only works when the whole line is a string, but here we have a `* 100` int at the end.

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 3]
```diff
@@ -267,6 +260,12 @@ def init_ddp_connection(self, global_rank: Optional[int], world_size: Optional[i
             log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
+            # on rank=0 let everyone know training is starting
+            rank_zero_info("-" * 100)
+            rank_zero_info(f"distributed_backend={self.distributed_backend}")
+            rank_zero_info(f"All DDP processes registered. Starting ddp with {self.world_size} processes")
+            rank_zero_info("-" * 100)
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:267]:
```suggestion
            rank_zero_info("-" * 100, 
            f"distributed_backend={self.distributed_backend}", 
            f"All DDP processes registered. Starting ddp with {self.world_size} processes", 
            "-" * 100))
```

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 4]
```diff
@@ -308,6 +301,12 @@ def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Opt
             log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
+            # on rank=0 let everyone know training is starting
+            rank_zero_info((
+                "-" * 100, f"distributed_backend={self.distributed_backend}",
+                f"All DDP processes registered. Starting ddp with {self.world_size} processes", "-" * 100
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/ddp.py:307]:
```suggestion
                f"All DDP processes registered. Starting ddp with {self.world_size} processes", "-" * 100,
```

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp.py]: made the choice to actually print the torch backend here. from the context, it makes much more sense. accelerator didn't seem right because we are dealing with plugins here.

**Comment by awaelchli** [comment]
@Borda the printing with tuple you suggested wasn't working because it would print everything on one line. I made use of f-strings now.

**Comment by Borda** [comment]
> @Borda the printing with tuple you suggested wasn't working because it would print everything on one line. I made use of f-strings now.

so just add line end, for example `os.linesep`

File: pytorch_lightning/plugins/training_type/ddp.py [hunk 5]
```diff
@@ -308,6 +301,14 @@ def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Opt
             log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
+            # on rank=0 let everyone know training is starting
+            rank_zero_info(
+                f"{'-' * 100}\n"
+                f"distributed_backend={self.torch_distributed_backend}\n"
+                f"All DDP processes registered. Starting ddp with {self.world_size} processes\n"
+                f"{'-' * 100}\n"
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/ddp.py:309]:
```suggestion
                f"{'-' * 100}{os.linesep}"
                f"distributed_backend={self.torch_distributed_backend}{os.linesep}"
                f"All DDP processes registered. Starting ddp with {self.world_size} processes{os.linesep}"
                f"{'-' * 100}{os.linesep}"
```

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 6]
```diff
@@ -267,6 +260,14 @@ def init_ddp_connection(self, global_rank: Optional[int], world_size: Optional[i
             log.info(f"initializing ddp: GLOBAL_RANK: {global_rank}, MEMBER: {global_rank + 1}/{world_size}")
             torch_distrib.init_process_group(self.torch_distributed_backend, rank=global_rank, world_size=world_size)
 
+            # on rank=0 let everyone know training is starting
+            rank_zero_info(
+                f"{'-' * 100}\n"
+                f"distributed_backend={self.torch_distributed_backend}\n"
+                f"All DDP processes registered. Starting ddp with {self.world_size} processes\n"
+                f"{'-' * 100}\n"
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:268]:
```suggestion
                f"{'-' * 100}{os.linesep}"
                f"distributed_backend={self.torch_distributed_backend}{os.linesep}"
                f"All DDP processes registered. Starting ddp with {self.world_size} processes{os.linesep}"
                f"{'-' * 100}{os.linesep}"
```

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:268]: what does this do differently? it seems unnecessary.

Comment (Borda) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:268]: Sometimes the /n doesn't work on Windows

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:268]: From
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

> In text mode, the default when reading is to convert platform-specific line endings (\n on Unix, \r\n on Windows) to just \n. When writing in text mode, the default is to convert occurrences of \n back to platform-specific line endings.

So we should not use `os.linesep` and let Python handle it.
